Makefile

```diff
@@ -500,36 +491,51 @@ test/karma-conf-dev.js: test/karma-conf.mako.js ${MAKO_CMD}
 test/karma-conf-prod.js: test/karma-conf.mako.js ${MAKO_CMD}
 	${PYTHON_CMD} ${MAKO_CMD} --var "mode=prod" $< > $@
 
-node_modules: ANGULAR_JS = angular.js angular.min.js
-node_modules: ANGULAR_TRANSLATE_JS = angular-translate.js angular-translate.min.js
-node_modules: ANGULAR_TRANSLATE_LOADER_JS = angular-translate-loader-static-files.js angular-translate-loader-static-files.min.js
-node_modules: LOCALFORAGE = localforage.js localforage.min.js
-node_modules: JQUERY = jquery.js jquery.min.js
-node_modules: JQUERYXDOMAIN = jQuery.XDomainRequest.js  jquery.xdomainrequest.min.js
-node_modules: D3 = d3.js  d3.min.js
-node_modules: BOOTSTRAP = bootstrap.js bootstrap.min.js
-node_modules: SLIPJS = slip.js
-node_modules: package.json
-	npm install
-	cp $(addprefix node_modules/angular/,$(ANGULAR_JS)) src/lib/;
-	cp $(addprefix node_modules/angular-translate/dist/,$(ANGULAR_TRANSLATE_JS)) src/lib/;
-	cp $(addprefix node_modules/angular-translate/dist/angular-translate-loader-static-files/,$(ANGULAR_TRANSLATE_LOADER_JS)) src/lib/;
-	cp $(addprefix node_modules/localforage/dist/,$(LOCALFORAGE)) src/lib/;
-	cp $(addprefix node_modules/slipjs/,$(SLIPJS)) src/lib;
-	cp $(addprefix node_modules/jquery/dist/,$(JQUERY)) src/lib/;
-	cp $(addprefix node_modules/jquery-ajax-transport-xdomainrequest/,$(JQUERYXDOMAIN)) src/lib/;
-	cp $(addprefix node_modules/d3/,$(D3)) src/lib/;
-	cp $(addprefix node_modules/bootstrap/dist/js/,$(BOOTSTRAP)) src/lib/;
-	cp node_modules/fastclick/lib/fastclick.js src/lib/;
-	cp node_modules/angular-mocks/angular-mocks.js test/lib/;
-	cp node_modules/expect.js/index.js test/lib/expect.js;
-	cp node_modules/sinon/pkg/sinon.js test/lib/;
-	cp node_modules/google-closure-compiler/contrib/externs/angular-1.4.js externs/angular.js;
-	cp node_modules/google-closure-compiler/contrib/externs/jquery-1.9.js externs/jquery.js;
+test/lib/angular-mocks.js:
+	cp -f node_modules/angular-mocks/angular-mocks.js $@;
+test/lib/expect.js:
+	cp -f node_modules/expect.js/index.js $@;
+test/lib/sinon.js:
+	cp -f node_modules/sinon/pkg/sinon.js $@;
+externs/angular.js:
+	cp -f node_modules/google-closure-compiler/contrib/externs/angular-1.4.js $@;
+externs/jquery.js:
+	cp -f node_modules/google-closure-compiler/contrib/externs/jquery-1.9.js $@;
+.PHONY: package_json_dev
+package_json_dev:
+	npm install --only=dev
+node_modules: package_json_dev externs/jquery.js externs/angular.js test/lib/sinon.js test/lib/expect.js test/lib/angular-mocks.js
+
+ANGULAR_JS = angular.js angular.min.js
+ANGULAR_TRANSLATE_JS = angular-translate.js angular-translate.min.js
+ANGULAR_TRANSLATE_LOADER_JS = angular-translate-loader-static-files.js angular-translate-loader-static-files.min.js
+LOCALFORAGE = localforage.js localforage.min.js
+JQUERY = jquery.js jquery.min.js
+JQUERYXDOMAIN = jQuery.XDomainRequest.js  jquery.xdomainrequest.min.js
+D3 = d3.js  d3.min.js
+BOOTSTRAP = bootstrap.js bootstrap.min.js
+SLIPJS = slip.js
```

Comment:
All those variables above are used exactly once below. We could safe some lines putting the filenames below directly.